webserver/static/scripts/datasets/editor.js

```diff
@@ -275,12 +275,21 @@ var DatasetControlButtons = React.createClass({
         if (this.props.mode == MODE_EDIT) {
             buttonText = "Update";
         }
+        if (this.state.errorMsg) {
+            var errorMessage = "Unknown error";
+            if (typeof this.state.errorMsg.error !== 'undefined') {
```

Comment:
I'm not sure if this is the right way of seeing if an object has a propery/field

Reply:
I would do this:

```
var error = null;
if (this.state.errorMsg) {
  error = <p className='text-danger'>
                    <strong>An error occurred while submitting this dataset:</strong>
                    &nbsp;{ this.state.errorMsg.error ||  "Unknown error"}
                </p>
}
```

for succinctness, but the current PR will work just fine.
By having error as `null` by default, React won't render any node instead of a node with an empty string.

Reply:
Thanks for the recommendation on making the code more readable. A review approval would be appreciated!